serverless/serverless
# update aws-java-gradle and aws-java-maven templates

- update the Java templates to work with lambda proxy integration
out-of-the-box
- use "Map<String, Object>" as input type for greater flexibility
- update gradle wrapper from 3.1 to 3.2.1
- configure gradle and maven to use Java 1.8
- change the package from "hello" to "com.serverless" and adjust
serverless.yml accordingly
- configure logging (log4j)
- adjust tests
- git-ignore some Eclipse specific files

## What did you implement:

Closes #2938 

Also see: http://forum.serverless.com/t/api-gateway-inst-created-ok-in-java-project/551

Updates the aws-java-gradle and aws-java-maven templates to work with lambda proxy integration out-of-the-box.

## How did you implement it:


## How can we verify it:

```bash
mkdir serverless-gradle
cd serverless-gradle
<slsdevdir>/bin/serverless create --template aws-java-gradle --name serverless-gradle
./gradlew build
<slsdevdir>/bin/serverless deploy
<slsdevdir>/bin/serverless invoke -f hello
```

```bash
mkdir serverless-maven
cd serverless-maven
<slsdevdir>/bin/serverless create --template aws-java-maven --name serverless-maven
mvn package # make sure you install maven; it's easiest to install maven through [sdkman](http://sdkman.io/) using "sdk install maven" (yes, use as default)
<slsdevdir>/bin/serverless deploy
<slsdevdir>/bin/serverless invoke -f hello
```

Change serverless.yml in both created projects (serverless-gradle and serverless-maven) and uncomment
```yml
    events:
      - http:
          path: users/create
          method: get
``` 
re-deploy it and curl the endpoints

## Todos:



***Is this ready for review?:*** YES
